jobs/cache_maintenance/src/cache_maintenance/backfill.py

```diff
@@ -69,39 +73,67 @@ def backfill_retryable_errors(
     )
 
 
+def try_backfill_dataset(
+    dataset: str,
+    hf_endpoint: str,
+    blocked_datasets: list[str],
+    hf_token: Optional[str] = None,
+    storage_clients: Optional[list[StorageClient]] = None,
+) -> BackfillStatistics:
+    try:
+        return BackfillStatistics(
+            num_analyzed_datasets=1,
+            operations=backfill_dataset(
+                dataset=dataset,
+                hf_endpoint=hf_endpoint,
+                blocked_datasets=blocked_datasets,
+                hf_token=hf_token,
+                priority=Priority.LOW,
+                hf_timeout_seconds=None,
+                storage_clients=storage_clients,
+            ),
+        )
+
+    except Exception as e:
+        logging.warning(f"failed to update_dataset {dataset}: {e}")
+        return BackfillStatistics(num_analyzed_datasets=1, num_error_datasets=1)
+
+
 def backfill_datasets(
     dataset_names: set[str],
     hf_endpoint: str,
     blocked_datasets: list[str],
     hf_token: Optional[str] = None,
     storage_clients: Optional[list[StorageClient]] = None,
-) -> None:
+) -> BackfillStatistics:
     logging.info(f"analyzing {len(dataset_names)} datasets in the database")
+
     statistics = BackfillStatistics(num_total_datasets=len(dataset_names))
-    log_batch = 100
-
-    for dataset in dataset_names:
-        try:
-            statistics.add(
-                BackfillStatistics(
-                    num_analyzed_datasets=1,
-                    operations=backfill_dataset(
-                        dataset=dataset,
-                        hf_endpoint=hf_endpoint,
-                        blocked_datasets=blocked_datasets,
-                        hf_token=hf_token,
-                        priority=Priority.LOW,
-                        hf_timeout_seconds=None,
-                        storage_clients=storage_clients,
-                    ),
+
+    with concurrent.futures.ThreadPoolExecutor(max_workers=MAX_BACKFILL_WORKERS) as executor:
+        futures_to_dataset = {
+            executor.submit(
```

Comment:
It looks like the `future` can be a dictionary key, but note that weird objects in python may not be hashable

Reply:
this structure comes from the official docs pages, so I felt free to use it in our code too

Reply:
anyway, we could remove it, it's just used for a warning log